### DISCUSSION TITLE
Saiba como Concluir o seu Desafio de Projeto na DIO

### DISCUSSION THREAD

**Discussion by elidianaandrade** [digitalinnovationone/dio-lab-open-source#28015] [Desafios DIO]
# Saiba como Concluir o seu Desafio de Projeto na DIO
**Os Desafios de Projeto na [DIO (Digital Innovation One)](https://www.dio.me/)** são essenciais para solidificar o aprendizado do aluno, auxiliando no desenvolvimento de habilidades práticas. De maneira geral, os Desafios de Projeto (Labs) são identificados como "**Projeto Entregue**" após o estudante enviar a URL do repositório presente em seu GitHub, contendo a resolução do desafio proposto pelo instrutor do Lab. Após isso, o seu projeto poderá ser visualizado na aba "**Projetos**" no seu perfil na DIO.
___

Para entendermos como enviar o Desafio de Projeto, vou tomar como exemplo o Desafio "**Contribuindo em um Projeto Open Source no GitHub**".

## 1. Assista ao Lab e Leia as instruções em "Entendendo o Desafio"
Assista ao Lab e certifique-se de ler atentamente as instruções do desafio na aba "Entendendo o Desafio". Nela, você poderá encontrar detalhes importantes sobre o que deve ser incluído no projeto e como ele deve ser apresentado. Além disso, a aba contém materiais de apoio, como links da apresentação e do repositório fornecido pelo instrutor.

![Aba Entendendo o Desafio no Desafio de Projeto da DIO](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/abd366f4-da34-4688-91aa-a3b5a13d7c8e)

## 2. Resolva o Desafio
Crie um novo repositório no GitHub ou faça um fork do repositório fornecido pelo instrutor com as alterações propostas no Desafio. Aproveite para usar a criatividade, adicionando novos incrementos.

> No caso do desafio  "**Contribuindo em um Projeto Open Source no GitHub**", você deverá enviar o seu fork do repositório contendo o seu perfil README na pasta "[community](https://github.com/digitalinnovationone/dio-lab-open-source/tree/main/community)".
> **Atenção:** Para que possa enviar o desafio, **não** é necessário que seu Pull Request seja aceito, tendo em vista que você criou o commit no seu repositório, ele estará presente lá independentemente de aceitar ou não o Pull Request. Ao aceitar o que acontece é que o seu commit passa a fazer parte do repositório `https://github.com/digitalinnovationone/dio-lab-open-source/` também.

## 3. Entregue o Projeto
1. Para enviar o seu Projeto, clique no botão "**Entregar Projeto**":
  ![Botão Entregar Projeto](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/3aa60be7-ff58-4917-872d-c7ca2d012d76)

2. **Preencha o Pop-up com as informações referentes ao seu projeto**, como URL e uma breve descrição. Sinalize também que está de acordo com os "Termos de Uso", para que o projeto possa ser entregue ao clicar no botão "**Entregar**":
![Pop-up Entrega do Desafio](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/1d03f37b-23a5-4a47-9968-c2053d7740b2)
No lab em questão, você deve enviar a URL do seu "fork" feito do repositório `https://github.com/digitalinnovationone/dio-lab-open-source` 

> Por exemplo, se o seu usuário for "elidianaandrade" e o nome do seu repositório "dio-lab-open-source", deverá enviar:
> ```
> https://github.com/elidianaandrade/dio-lab-open-source/
> ```

3. **Feito isso, o seu desafio será entregue com sucesso** ✅ Você poderá visualizá-lo na aba "**Projetos**" no seu perfil da DIO:
![Aba "Projetos" no Perfil da DIO](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/f38adf15-96e0-48d2-9902-2eb63bffe212)

> Você poderá acessar através do link abaixo, substituindo SEUUSUARONADIO pelo seu nome de usuário na Plataforma:
> ```
> https://web.dio.me/users/SEUUSUARONADIO?tab=projects&page=1
> ```

## Como Atualizar o seu Desafio
1. Caso queira, poderá atualizar as informações presentes em um desafio já enviado, como a URL e a descrição. Para isso, basta acessar a página do Desafio e clicar no botão "**Projeto Entregue**":
![Botão Projeto Entregue](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/a3ff5411-9132-4e3f-b9b0-097058164dc2)

2. Feito isso, o Pop-up será aberto novamente, mas dessa vez para Atualização do Projeto. Após atualizar, basta clicar no botão "**Atualizar**" e as suas modificações serão salvas:
![Pop-up Atualização do Projeto](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/369f58b4-05cc-456e-82a4-cccb624ee069)

___

## Saiba Mais
- Curso: Desafios de Projetos: Crie Um Portfólio Vencedor

**Comment by joaojanuario19** [comment]
Elidiana meu PR deu erro, porque ao invés joaojanuario19.md como nome do arquivo, eu coloquei joaojanuario.md, está dando esse erro. preciso de ajuda para corrigir
![Captura de tela 2024-05-27 142305](https://github.com/digitalinnovationone/dio-lab-open-source/assets/108444902/b1ef1dcc-b4ff-4dc6-b49a-b369a7785566)
![Captura de tela 2024-05-27 142321](https://github.com/digitalinnovationone/dio-lab-open-source/assets/108444902/f79db27c-08c7-4220-85b0-afddb18f379d)

**Reply by elidianaandrade** [reply]
Olá, @joaojanuario19! Tudo bem? Massa! Vi aqui que já corrigiu o seu PR 👏👏, mas respondendo caso alguém tenha o mesmo problema, basta renomear o arquivo para o seu nome de usuário. Além disso, certifique-se também se o arquivo está inserido na pasta "community". Parabéns e bons estudos! 😊👏👏

**Comment by esmaelsena** [comment]
olá Elidiana, cheguei nessa parte do curso agora, da entrega do primeiro projeto, e não sei se estou procurando no lugar certo rsrs!! mas não estou encontrando o repositorio do projeto.

**Reply by elidianaandrade** [reply]
Olá @esmaelsena! Você está no repositório certo :) é esse aqui [https://github.com/digitalinnovationone/dio-lab-open-source](https://github.com/digitalinnovationone/dio-lab-open-source). Ele foi transferido aqui para a conta da DIO.

**Comment by PCVCPAULO** [comment]
Olá! Boa noite Elidiana! Espero que estejas bem! Na vídeo aula "Desvendando uma Contribuição", no minuto 17:09 aproximadamente, quando você executa o cmando "git status" aparece os arquivos "untracked". No meu caso foi o seguinte, apareceu um arquivo deleted.000atuan.md. Eu executei o comando "git add community/#pcvcpaulo.md", que era o meu arquivo de contribuição, porém quando executei o comando 'git commit -m "feat: add pcvcpaulo profile", foi considerado o meu arquivo "#pcvcpaulo.md" e o arquivo "000atuan.md" que não sei quem é. Dando sequência em sua aula, executei o comando git push origin feat/community/pcvpaulo que enviou as minhas alterações, porém quando consultei o repositório remoto,o tal arquivo "000atuan.md" foi junto com o meu arquivo "#pcvcpaulo.md", travando todo o restante dos procedimentos e não atualizando o repositório remoto. 

No meu "fork" recebi vários retorno com o erro, inclusive no meu e-mail também chegou mensagens avisando sobre o acontecido.

Vou realizar mais uma tentativa, mas se gerar o mesmo erro, vou ter que desisti o bootcamp.

**Reply by elidianaandrade** [reply]
Olá @PCVCPAULO! Bom dia, tudo bem? Sobre o que disse inicialmente vou verificar, mas em relação ao seu Pull Request ele chegou (veja em #38463), a automação só não mesclou, pois você nomeou como "#pcvcpaulo" e precisa ser idêntico ao seu nome de usuário, nesse caso **PCVCPAULO**, tudo em maiúsculo e sem hashtag. 

> ![image](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/08fa789b-4757-4b86-815c-9a86ea1b0a09)
> ![image](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/027f8cf8-1015-4c38-ac95-a387b46690a0)

**Você pode renomear pelo próprio navegador, seguindo os passos:**
1. Acesse seu Pull #38463
2. No canto direito, vá na aba "Files changed" e clique nos "três pontinhos" e clique em **Edit File**, conforme mostro na imagem abaixo:

> ![image](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/6ade9b24-14bd-4ce4-a8d1-7f72b94a07ee)

3. Altere o nome para PCVCPAULO.md e commit as alterações.

> ![image](https://github.com/digitalinnovationone/dio-lab-open-source/assets/97471199/2d5b03bd-e593-453a-b484-3ddab3161b8b)

Espero ter ajudado! Bons estudos! E não desista :)

**Reply by PCVCPAULO** [reply]
Boa tarde Eli!
Muito obrigado pelo retorno.
Depois de suas orientações consegui conclui o desafio.

**Comment by Kmionip** [comment]
Muito obrigado à todos pelas aulas e pela paciência.

**Comment by arafasilva** [comment]
Olá! Comecei o bootcamp nesta semana e estou na parte do desafio do Projeto Open Source. Consegui clonar no meu git mas no momento em que vou criar uma new branch aparece a seguinte mensagem: "fatal: not a git repository (or any of the parent sirectories): .git"

Não sei aonde errei aqui!rsrs

**Reply by AdaoElias** [reply]
O erro "fatal: not a git repository (or any of the parent directories): .git" indica que você está tentando executar um comando Git em um diretório que não é um repositório Git válido. Isso geralmente acontece quando você não está dentro de um repositório Git ou quando o diretório atual não contém a pasta .git, que é necessária para que o Git reconheça o repositório. Para resolver isso, você pode verificar se está no diretório correto ou inicializar um novo repositório usando git init.

**Reply by mayariam** [reply]
Aconteceu isso comigo também.

**Comment by oMaestro174** [comment]
❤️

**Comment by FernandaPortoAlegre** [comment]
Olá, boa tarde!
não estou conseguindo realizar o desafio, primeiro eu preciso criar um repositório no github para iniciar?
o meu está dando vários erros.

**Comment by FernandaPortoAlegre** [comment]
Olá Elidiana, o meu deu erro igual do Paulo que informou no comentário acima. Já tentei alterar o meu nome mas não está dando certo. Me ajuda?!

[validate-and-comment](https://github.com/digitalinnovationone/dio-lab-open-source/actions/runs/11021372279/job/30608292583?pr=48982#logs)
failed 3 minutes ago in 8s
Search logs
1s
3s
1s
Run AUTHOR_NAME="FernandaPortoAlegre"
Changed files:
community/fernandaportoalegre.md
ERROR: Invalid PR. More or different files than expected were changed.
Error: Process completed with exit code [1](https://github.com/digitalinnovationone/dio-lab-open-source/actions/runs/11021372279/job/30608292583?pr=48982#step:3:1).
0s
1s
0s

**Comment by Thiagollalves** [comment]
Olá estou tendo sempre esse erro. O que eu faço ? Eu não consigo.  @elidianaandrade 


![image](https://github.com/user-attachments/assets/42a15250-a87f-4e11-ac70-e92406e3c1fd)
![Captura de tela 2024-10-10 163433](https://github.com/user-attachments/assets/d798ba0b-f115-4676-914f-2d1e92767566)
![image](https://github.com/user-attachments/assets/3b380250-107c-416e-9554-e74441181776)
![image](https://github.com/user-attachments/assets/255b3a66-70ed-4a39-acb5-d10422a247a1)

**Comment by Thiagollalves** [comment]
Aparecem dois arquivos. É o correto?

![image](https://github.com/user-attachments/assets/0c6ff8b0-e1e0-42e8-a8b1-267d2294d4d8)

**Comment by LucasMDejane** [comment]
Olá estou com problemas no meu pull request, apaguei ele e vou até tentar mais uma vez tentar de novo. Porém aparece que eu modifiquei arquivo de uma pessoa aleatória, mas nunca mexi em outro arquivo. segui seus videos e fui fazendo do jeito certinho idêntico. Apenas criei meu próprio ReadMe. Já tentei por várias soluções, e não acho.

![imagem_2025-06-28_004303781](https://github.com/user-attachments/assets/57ca6c3e-3fe7-44a6-a0ac-21f0ad35ea5b)
